<a href="https://colab.research.google.com/github/ArnavP2305/flyrank-ml-internship-starter/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

# Lane: Refresh / Content Opportunity Scoring

**I chose this lane because the dataset contains search performance metrics such as impressions, clicks, CTR, average position, freshness, engagement, and trend information. These signals can help identify content that is underperforming and should be reviewed first. Rather than manually inspecting thousands of pages, a scoring system can prioritize pages that are likely to benefit from updates.**

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

# Research Question

**Which webpages should be prioritized for content refresh based on their search performance and engagement metrics?**

# Decision

**Decide which pages should be reviewed and updated first.**

# Action

**The SEO/content team can refresh titles, update outdated content, improve metadata, strengthen internal linking, or expand the article.**

# Cost of Wrong Call

**If the wrong pages are prioritized, the team spends time updating pages with little potential while overlooking pages that could generate meaningful search traffic improvements.**

# Why ML or Data Helps

**Manually reviewing thousands of pages is difficult. By combining multiple signals such as CTR, impressions, trend, freshness, and engagement, a scoring system can consistently rank pages that deserve attention first.**

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [10]:
import pandas as pd

df = pd.read_csv("content_refresh_anonymized.csv")

print("Total pages:", len(df))
print("Average CTR:", round(df["ctr"].mean(), 2))
print("Average impressions:", round(df["impressions_90d"].mean(), 2))
print("Pages with CTR < 1%:", (df["ctr"] < 1).sum())
print(
    "Pages with CTR < 1% and impressions > 1000:",
    ((df["ctr"] < 1) & (df["impressions_90d"] > 1000)).sum()
)

Total pages: 30000
Average CTR: 0.51
Average impressions: 5200.37
Pages with CTR < 1%: 28291
Pages with CTR < 1% and impressions > 1000: 13030


**The starter dataset contains 30,000 webpages, providing a sufficiently large sample for analysis. The average CTR is 0.51%, indicating that many pages receive relatively few clicks compared to their impressions. Additionally, 28,291 pages have a CTR below 1%, and 13,030 pages have both a CTR below 1% and more than 1,000 impressions. These observations suggest that there are many pages with visibility but relatively low engagement, making content refresh prioritization a meaningful research direction.**

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

**This project identifies pages that appear to have high refresh potential based on historical search and engagement data. It does not guarantee that refreshing content will improve rankings or traffic because search performance also depends on user intent, competition, and search engine algorithms. The recommendations should therefore be treated as decision support rather than definitive predictions.**

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.